src/main/kotlin/ro/hibyte/betting/entity/BetTemplate.kt

```diff
@@ -0,0 +1,72 @@
+package ro.hibyte.betting.entity
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import jakarta.persistence.*
+import ro.hibyte.betting.exceptions.types.BadRequestException
+import ro.hibyte.betting.repository.BetTemplateRepository
+
+
+@Entity
+data class BetTemplate (
+    @Id
+    @GeneratedValue
+    var id: Long? = null,
+
+    var name: String,
+
+    @Enumerated(EnumType.STRING)
+    var type: BetTemplateType,
+
+    @JsonIgnore
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "betTemplate", cascade = [CascadeType.ALL])
+    var betTypes: List<BetType> = ArrayList(),
+
+    @ElementCollection
+    @CollectionTable(name = "bet_template_options", joinColumns = [JoinColumn(name = "bet_template_id")])
+    @Column(name = "option")
+    var multipleChoiceOptions: List<String> = ArrayList()
+
+) {
+
+
+    companion object {
+        //this method ensures that no MULTIPLE_CHOICE BetTemplates can be created if the number of options is less than 2
```

Comment:
I feel like this comment is not needed as the method's signature is pretty self-explainatory. Personally I'm not a huge fan of comments as I think that more often than not they bring more noise to the codebase than they do good. [Here](https://youtu.be/Bf7vDBBOBUA?si=GHXLNYDNX-3PtkMu)'s a video/channel that I really like that explains what I mean.